src/Microsoft.Health.Fhir.SqlServer/Features/Schema/Migrations/8.diff.sql

```diff
@@ -922,3 +922,494 @@ AS
+(
+    ResourceTypeId smallint NOT NULL,
+    ResourceId varchar(64) COLLATE Latin1_General_100_CS_AS NOT NULL,
+    SearchParamId smallint NOT NULL,
+    SystemId1 int NULL,
+    Code1 varchar(128) COLLATE Latin1_General_100_CS_AS NOT NULL,
+    Text2 nvarchar(256) COLLATE Latin1_General_100_CI_AI_SC NOT NULL,
+    TextOverflow2 nvarchar(max) COLLATE Latin1_General_100_CI_AI_SC NULL
+)
+
+CREATE TYPE dbo.BulkTokenNumberNumberCompositeSearchParamTableType_1 AS TABLE
+(
+    ResourceTypeId smallint NOT NULL,
+    ResourceId varchar(64) COLLATE Latin1_General_100_CS_AS NOT NULL,
+    SearchParamId smallint NOT NULL,
+    SystemId1 int NULL,
+    Code1 varchar(128) COLLATE Latin1_General_100_CS_AS NOT NULL,
+    SingleValue2 decimal(18,6) NULL,
+    LowValue2 decimal(18,6) NULL,
+    HighValue2 decimal(18,6) NULL,
+    SingleValue3 decimal(18,6) NULL,
+    LowValue3 decimal(18,6) NULL,
+    HighValue3 decimal(18,6) NULL,
+    HasRange bit NOT NULL
+)
+
+CREATE TYPE dbo.BulkReindexResourceTableType_1 AS TABLE
+(
+    ResourceTypeId smallint NOT NULL,
+    ResourceId varchar(64) COLLATE Latin1_General_100_CS_AS NOT NULL,
+    ETag int NULL,
+    SearchParamHash varchar(64) NOT NULL
+)
+
+GO
+
+--
+-- STORED PROCEDURE
+--     BulkReindexResources
+--
+-- DESCRIPTION
+--     Updates the search indices of a batch of resources
+--
+-- PARAMETERS
+--     @resourcesToReindex
+--         * The type IDs, IDs, eTags and hashes of the resources to reindex
+--     @resourceWriteClaims
+--         * Claims on the principal that performed the write
+--     @bulkCompartmentAssignments
+--         * Compartments that the resource is part of
+--     @bulkReferenceSearchParams
+--         * Extracted reference search params
+--     @bulkTokenSearchParams
+--         * Extracted token search params
+--     @bulkTokenTextSearchParams
+--         * The text representation of extracted token search params
+--     @bulkStringSearchParams
+--         * Extracted string search params
+--     @bulkNumberSearchParams
+--         * Extracted number search params
+--     @bulkQuantitySearchParams
+--         * Extracted quantity search params
+--     @bulkUriSearchParams
+--         * Extracted URI search params
+--     @bulkDateTimeSearchParms
+--         * Extracted datetime search params
+--     @bulkReferenceTokenCompositeSearchParams
+--         * Extracted reference$token search params
+--     @bulkTokenTokenCompositeSearchParams
+--         * Extracted token$token tokensearch params
+--     @bulkTokenDateTimeCompositeSearchParams
+--         * Extracted token$datetime search params
+--     @bulkTokenQuantityCompositeSearchParams
+--         * Extracted token$quantity search params
+--     @bulkTokenStringCompositeSearchParams
+--         * Extracted token$string search params
+--     @bulkTokenNumberNumberCompositeSearchParams
+--         * Extracted token$number$number search params
+--
+CREATE PROCEDURE dbo.BulkReindexResources
```

Comment:
@johnstairs I created [AB#80109](https://microsofthealth.visualstudio.com/Health/_workitems/edit/80109) to investigate the potential performance gains of using `MERGE` instead of `DELETE` + `INSERT`.